yeonjae/programmers-1/숫자문자열과영단어.js

```diff
@@ -0,0 +1,21 @@
+function solution(s) {
+  const numArr = [
+    "zero",
+    "one",
+    "two",
+    "three",
+    "four",
+    "five",
+    "six",
+    "seven",
+    "eight",
+    "nine",
+  ];
+  let result = s;
+
+  for (let i = 0; i < numArr.length; i++) {
+    let array = result.split(numArr[i]);
+    result = array.join(i);
+  }
+  return Number(result);
+}
```

Comment:
```suggestion
function solution(s) {
  const numArr = [
    "zero",
    "one",
    "two",
    "three",
    "four",
    "five",
    "six",
    "seven",
    "eight",
    "nine",
  ];
  let result = s;

  for (let i = 0; i < numArr.length; i++) {
    const array = result.split(numArr[i]);
    result = array.join(i);
  }
  return Number(result);
}
```
- 풀이 잘봤씁니다! 🙇‍♂️
- `split`, `join` 특성을 잘 이해하시고 푸셨네요! 어떤 원리로 작동하는지 완전히 이해하셨을까요? 😮
- 이번주에 올라오는 풀이들이 어떤 원리로 작동 하는지에대한 풀이 내용이 없는데 풀이 설명도 주석으로 작성해주세요
- 참고로 array 변수는 재할당하는게 아니니 `const`로 선언해주세요 :) 
- 아래는 제 풀이입니다 정규표현식을 활용하였습니다 🙇‍♂️
```js
function solution(s) {
  const numberMappingObj = {
    zero: 0,
    one: 1,
    two: 2,
    three: 3,
    four: 4,
    five: 5,
    six: 6,
    seven: 7,
    eight: 8,
    nine: 9,
  };
  const regex = /(zero|one|two|three|four|five|six|seven|eight|nine)/g;

  return +s.replace(regex, (el) => numberMappingObj[el]);
}
```

Reply:
정규표현식 풀이 참고해보겠습니다! 풀이 추가로 올렸어요 :)